test/GithubApi.Contract.test.js

```diff
@@ -0,0 +1,25 @@
+const agent = require('superagent-promise')(require('superagent'), Promise);
+const chai = require('chai');
+const { listPublicEventsSchema } = require('./schema/ListPublicEvents.schema');
+
+const { expect } = chai;
+chai.use(require('chai-json-schema'));
+
+const urlBase = 'https://api.github.com';
+
+describe('Given event Github API resources', () => {
+  describe('When wanna verify the List public events', () => {
+    let listPublicEventsQuery;
+
+    before(() => {
+      listPublicEventsQuery = agent
+        .get(`${urlBase}/events`)
+        .auth('token', process.env.ACCESS_TOKEN);
+    });
+
+    it('then the body should have a schema', () =>
+      listPublicEventsQuery.then((response) => {
```

Comment:
si puedes hacer el then en el before es mejor hacerlo y dejarle las validaciones al it (en casos muy especificos puede ser valida la aproximacion que tienes, pero no para este caso)